frontend/src/components/note/NoteCard.vue

```diff
@@ -0,0 +1,102 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+
+import { Card, Divider } from '@/lib/primevue';
+import { userService } from '@/services';
+import { formatDateShort } from '@/utils/formatters';
+
+import type { Ref } from 'vue';
+import type { Note } from '@/types';
+
+// Props
+type Props = {
+  note: Note;
+  submissionId: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+// State
+const userName: Ref<string> = ref('');
+
+onMounted(async () => {
+  if (props.note.createdBy) {
+    const res = (await userService.searchUsers({ identityId: [props.note.createdBy] })).data;
+    userName.value = res.length ? res.pop().fullName : '';
```

Comment:
Is there a reason why we're using `Array.prototype.pop()` to acquire the last element's fullName (which modifies the res object by also dropping that last element) instead of a less impactful read-only `array.slice(-1)`?